### DISCUSSION TITLE
Pimcore Studio Preview

### DISCUSSION THREAD

**Discussion by fashxp** [pimcore/pimcore#17076] [General]
Dear Pimcore Community,

We are excited to share updates on the new Pimcore Admin UI, now called **Pimcore Studio**. During our last Pimcore Inspire event, we offered an initial glimpse into the UX design and the current state of development. We also committed to providing preview access to all code repositories when the time is right.

In that session, we detailed our concepts, approach, rationale, and roadmap. For a comprehensive understanding, we encourage you to watch [the video](https://youtu.be/hnOdy5dC88A?list=PLrlLr70ddFwI2jmKOHrnsLyZwUUZ-A4pm&t=1268) where we explain everything in detail.

Following you find some screens with the latest tweaks and updates: 

<img src="https://github.com/pimcore/pimcore/assets/8792145/43611819-935e-493b-b608-05fa98a0ed21" width="400" />
<img src="https://github.com/pimcore/pimcore/assets/8792145/54374c55-ea16-4172-848a-90472d3c9c3e" width="400" />
<img src="https://github.com/pimcore/pimcore/assets/8792145/b73e9aae-fe81-4716-8a81-6fcdd9b35dcd" width="400" />

<br><br>

After an exclusive preview period for selected partners, I'm happy to announce that **today is the day** we **open the repositories up to the public**.

### Following are the steps to install Pimcore Studio into a Pimcore instance: 
- Install and setup Generic Data Index Bundle, see https://pimcore.com/docs/platform/Generic_Data_Index/Installation/

- Install via composer 
```bash
composer require pimcore/studio-backend-bundle:1.x-dev pimcore/studio-ui-bundle:1.x-dev
```

- Activate following bundles in `bundles.php`
```php
Pimcore\Bundle\StudioBackendBundle\PimcoreStudioBackendBundle::class => ['all' => true],
Pimcore\Bundle\StudioUiBundle\PimcoreStudioUiBundle::class => ['all' => true],
```

- Run following installer: 
```bash
bin/console pimcore:bundle:install PimcoreStudioBackendBundle
```

- Login into Pimcore backend `https://<your-domain>/admin`
As soon as you are logged-in, you can access Pimcore Studio with `https://<your-domain>/admin/studio`
Also have a look at the `README` files in the repositories ([Backend README](https://github.com/pimcore/studio-backend-bundle/blob/1.x/README.md), [UI README](https://github.com/pimcore/studio-ui-bundle/blob/1.x/README.md)) for some additional information. 

Please keep in mind, that this is a **preview of the current development state and not a release**! All of this is under ongoing development and might be subject to changes as we proceed. 
Also since Inspire, there were already some changes - for example we decided not to use API Platform for the API due to numerous reasons. 

### Our next steps working forward will be
- Completing basics of Pimcore assets with all widgets
- Finalizing basic concepts and public frontend API 
- Implementing Pimcore data objects 

We are happy to get your feedback and findings and look forward to working with you on a successful and powerful new solution for Pimcore clients worldwide. Please direct your feedback directly via the github repository issues.  

Thanks for your support making Pimcore better every day and enjoy your day.

**Comment by kg303** [comment]
![Screenshot 2024-05-28 162543](https://github.com/pimcore/pimcore/assets/122809595/e8914b8f-e0e6-4003-b7b3-402a22164860)

Why can't I see anything else in studio? I have done everything from the installation guide.

**Reply by dpfaffenbauer** [reply]
there is nothing else? have you tried open a asset?

**Reply by kg303** [reply]
> there is nothing else? have you tried open a asset?

Yes, assets are available, but data objects, the menu, and the rest are missing. 
Is it just me, or are those functionalities not available yet?

**Reply by dpfaffenbauer** [reply]
I believe it is not available yet.

**Reply by markus-moser** [reply]
Those functionalities are not available yet. This is a very early development preview. It will definitely need quite a bit more time until we can provide the data object functionalities.

Take a look at the [the video](https://youtu.be/hnOdy5dC88A?list=PLrlLr70ddFwI2jmKOHrnsLyZwUUZ-A4pm&t=1268) for more details about the roadmap.

**Reply by kg303** [reply]
Thank you very much for quick replies.

**Comment by zieglerh** [comment]
@fashxp can you please explain why you decided agains API Platform and what the numerous reasons are? Would be good to know. Thanks 👍

**Reply by fashxp** [reply]
For us, the following aspects were the most important:

- The API platform provides many benefits when using Doctrine entities. Since Pimcore does not use them, we cannot benefit from these advantages.
- Creating custom endpoints, especially when you have specific requirements, turned out to be tricky and complex.
- Documentation and other resources for such custom requirements are very rare.
- Running multiple API platform instances in one Pimcore project seems not easily possible. A use case would be that a Pimcore implementation (or another bundle) already uses the API platform, which would then conflict with (or not be possible in) Pimcore Studio.

In general, using a plain Symfony controller approach, defining schemas and definitions via custom attributes, and using `zircote/swagger-php` for generating the OpenAPI documentation feels much more appropriate for our use case and provides a better developer experience.

Kudos to @solverat for providing valuable input and experience on this!

**Reply by dpfaffenbauer** [reply]
@fashxp we did a API Platform Integration with Pimcore DataObjects couple of years ago (https://github.com/instride-ch/pimcore-api-platform). It was quite tricky to be honest and I prefer a more "custom" approach now as well.

**Reply by solverat** [reply]
@dpfaffenbauer and as you can see in https://github.com/instride-ch/pimcore-api-platform/issues/4, the struggle is/was real. :) 

Using symfonys (powerful) core features to build robust api definitions instead of adding an additional layer of complexity is a good thing. Besides, if your fighting against api platform principals you'll lose the battle for sure.

**Comment by dpfaffenbauer** [comment]
Is there some kind of timeline when certain features will/should be ready? Like Documents or DataObjects?

**Reply by fashxp** [reply]
Timeline no, order of implementation will be
1) assets (should be more or less ready next two months)
2) data objects (hopefully end of year, we will be able to communicate a more precise timeline once we have the assets)
3) documents
4) everything else for editors
5) everything else for admin

**Reply by dpfaffenbauer** [reply]
regarding documents: any ideas how to implement that? I guess the datamodel will stay compatible and you support twig rendering?

**Reply by fashxp** [reply]
jup, documents concept will stay the same (at least for now), as we need to support admin-classic-ui and studio-ui in parallel for a while.

**Comment by hoangngx** [comment]


**Reply by jdreesen** [reply]
Did you add it to your `config/bundles.php` or kernel class?
